corehq/apps/data_dictionary/views.py

```diff
@@ -166,6 +172,15 @@ def deprecate_or_restore_case_type(request, domain, case_type_name):
     return JsonResponse({'status': 'success'})
 
 
+@login_and_domain_required
+@requires_privilege_with_fallback(privileges.DATA_DICTIONARY)
+@require_permission(HqPermissions.edit_data_dict)
+def delete_case_type(request, domain, case_type_name):
+    case_type_obj = CaseType.objects.get(domain=domain, name=case_type_name)
+    case_type_obj.delete()
```

Comment:
nit: do we want to check for missing case type here like we did in `delete_case_property`?

Reply:
This might be a good check to have. Although I don't foresee this scenario happening, there might be an unforeseen edge case and so we should rather stay on the cautious side.

Reply:
[adde543](https://github.com/dimagi/commcare-hq/pull/33956/commits/adde54370dd858ea0f4fae95198b583301221a57)